pkg/render/amazoncloudintegration.go

```diff
@@ -204,6 +205,9 @@ func (c *amazonCloudIntegrationComponent) credentialSecret() *corev1.Secret {
 func (c *amazonCloudIntegrationComponent) deployment() *appsv1.Deployment {
 	var replicas int32 = 1
 
+	annotations := make(map[string]string)
+	annotations[credentialSecretHashAnnotation] = AnnotationHash(c.credentials)
```

Comment:
I know you're just copying the design pattern used elsewhere, but I'm wondering if this will expose credentials.

A malicious user just needs RBAC that allows them to `get deployments`, then they can sha1 decrypt the annotation to get this secret.

Reply:
No, this creates a hash which is not decryptable.